src/main/java/com/insidious/plugin/ui/testrunnerinjection/TestRunnerInjector.java

```diff
@@ -0,0 +1,231 @@
+    private JPanel gradleWrapPanel;
+    private JLabel dependencyLink;
+
+    private String runnerCode = "import io.unlogged.runner.UnloggedTestRunner;\n" +
+            "import org.junit.runner.RunWith;\n\n" +
+            "@RunWith(UnloggedTestRunner.class)\n" +
+            "public class UnloggedTest {\n" +
+            "}";
+
+    //Services
+    private final InsidiousService insidiousService;
+    private final MultiModuleManager multiModuleManager;
+
+
+    /**
+     * Constructor for TestRunnerInjector.
+     * Initializes the UI components, sets up the module dropdown, and adds action listeners.
+     *
+     * @param insidiousService
+     */
+    public TestRunnerInjector(InsidiousService insidiousService) {
+        this.insidiousService = insidiousService;
+        this.multiModuleManager = new MultiModuleManager(insidiousService.getProject());
+
+        addBordersToTextBoxes();
+
+        testRunnerText.setText(runnerCode);
+        initializeModuleDropDown();
+        addInjectButtonActionListener();
+
+        //Copy Buttons
+
+        copyRunnerCode.addActionListener(e -> copyCode(CopyType.RUNNER));
+        mvnCopyButton.addActionListener(e -> copyCode(CopyType.MAVEN));
+        gradleCopyButton.addActionListener(e -> copyCode(CopyType.GRADLE));
+
+        //Open Documentation
+
+        link.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));
+        link.addMouseListener(new MouseAdapter() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                routeToCiDocumentation();
+            }
+        });
+        dependencyLink.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));
+        dependencyLink.addMouseListener(new MouseAdapter() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                routeToCiDocumentation();
+            }
+        });
+    }
+
+    /**
+     * Returns the main panel of the UI.
+     * @return the main JPanel
+     */
+    public JPanel getMainPanel() {
+        return mainPanel;
+    }
+
+    /**
+     * Copies the code to the clipboard based on the specified CopyType.
+     * @param type the type of code to copy (RUNNER, MAVEN, or GRADLE)
+     */
+    public void copyCode(CopyType type) {
+        if (type.equals(CopyType.RUNNER)) {
+            insidiousService.copyToClipboard(testRunnerText.getText());
+            InsidiousNotification.notifyMessage("Copied to clipboard",
+                    NotificationType.INFORMATION);
+        } else if(type.equals(CopyType.MAVEN)){
+            insidiousService.copyToClipboard(mvnTestTextArea.getText());
+            InsidiousNotification.notifyMessage("Copied to clipboard",
+                    NotificationType.INFORMATION);
+        } else {
+            insidiousService.copyToClipboard(gradleTestTextArea.getText());
+            InsidiousNotification.notifyMessage("Copied to clipboard",
+                    NotificationType.INFORMATION);
```

Comment:
Call to `notifyMessage()` can be moved outside of conditional blocks.

Reply:
will take it up with other changes in the revision